pkg/okteto/preview.go

```diff
@@ -0,0 +1,143 @@
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/machinebox/graphql"
+	"github.com/okteto/okteto/pkg/errors"
+)
+
+// DeployPreviewBody top body answer
+type DeployPreviewBody struct {
+	PreviewEnviroment PreviewEnv `json:"deployPreview" yaml:"deployPreview"`
+}
+
+// PreviewBody top body answer
+type PreviewBody struct {
+	Preview Preview `json:"preview"`
+}
+
+// Space represents the contents of an Okteto Cloud space
+type Preview struct {
+	GitDeploys []PipelineRun `json:"gitDeploys"`
+}
+
+//Previews represents an Okteto list of spaces
+type Previews struct {
+	Previews []PreviewEnv `json:"previews" yaml:"previews"`
+}
+
+//PreviewEnv represents an Okteto preview environment
+type PreviewEnv struct {
+	ID       string `json:"id" yaml:"id"`
+	Sleeping bool   `json:"sleeping" yaml:"sleeping"`
+	Scope    string `json:"scope" yaml:"scope"`
+}
+
+// CreatePreview creates a preview environment
+func DeployPreview(ctx context.Context, scope, repository, branch, sourceUrl, filename string, variables []Variable) (string, error) {
+	var body DeployPreviewBody
+	filenameParameter := ""
+	if filename != "" {
+		filenameParameter = fmt.Sprintf(`, filename: "%s"`, filename)
+	}
+
+	if len(variables) > 0 {
+		q := fmt.Sprintf(`mutation deployPreview($variables: [InputVariable]){
+			deployPreview(scope: %s, repository: "%s", branch: "%s", sourceUrl: "%s", variables: $variables%s){
+				id
+			},
+		}`, scope, repository, branch, sourceUrl, filenameParameter)
+
+		req := graphql.NewRequest(q)
+		req.Var("variables", variables)
+		if err := queryWithRequest(ctx, req, &body); err != nil {
+			if strings.Contains(err.Error(), "operation-not-permitted") {
+				return "", errors.UserError{E: fmt.Errorf("You are not authorized to create a global preview env."),
+					Hint: "Please log in with an administrator account or use a personal preview environment"}
+			}
+			return "", err
+		}
+	} else {
+		q := fmt.Sprintf(`mutation{
+			deployPreview(scope: %s, repository: "%s", branch: "%s"%s){
+				id
+			},
+		}`, scope, repository, branch, filenameParameter)
+
+		if err := query(ctx, q, &body); err != nil {
+			return "", err
+		}
+	}
+
+	return body.PreviewEnviroment.ID, nil
+}
+
+// DestroyPreview destroy a preview environment
+func DestroyPreview(ctx context.Context, branch, repository, scope string) error {
+	q := fmt.Sprintf(`mutation{
+		destroyPreview(scope: %s, repository: "%s", branch: "%s"){
+			id
+		},
+	}`, scope, repository, branch)
+
+	var body DeleteBody
+	return query(ctx, q, &body)
+}
+
+// ListPreviews list preview environments
+func ListPreviews(ctx context.Context) ([]PreviewEnv, error) {
+	q := `query{
+		previews{
+			id,
+			sleeping,
+			scope,
+		},
+	}`
+
+	var body Previews
+	if err := query(ctx, q, &body); err != nil {
+		return nil, err
+	}
+
+	return body.Previews, nil
+}
+
+// GetPreviewEnvByName gets a preview environment given its name
+func GetPreviewEnvByName(ctx context.Context, name, namespace string) (*PipelineRun, error) {
```

Comment:
are we using this method?

Reply:
yes, We are using that to check when the preview env is fully deployed